File: LeetCode/0944. Delete Columns to Make Sorted/solution.cpp
/**
 * 85 / 85 test cases passed.
 * Runtime: 36 ms
 * Memory Usage: 11.9 MB 
 */
class Solution {
public:
    int minDeletionSize(vector<string>& strs) {
        int n = strs[0].size();
        int ans = 0;
        for (int j = 0; j < n; ++ j) {
            for (int i = 1; i < strs.size(); ++ i) {
                if (strs[i][j] < strs[i - 1][j]) {
                    ++ ans;
                    break;
                }
            }
        }
        return ans;
    }
};

File: LeetCode/0944. Delete Columns to Make Sorted/solution.py
"""
85 / 85 test cases passed.
Runtime: 156 ms
Memory Usage: 15.7 MB
"""
class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        n = len(strs[0])
        rec = set()
        for i in range(1, len(strs)):
            for j in range(n):
                if j not in rec and strs[i][j] < strs[i - 1][j]:
                    rec.add(j)
            if len(rec) == n:
                return n
        return len(rec)
